lab_2_keywords_cooccurrence/main.py

```diff
@@ -3,12 +3,75 @@
 Extract keywords based on co-occurrence frequency
 """
 from pathlib import Path
-from typing import Optional, Sequence, Mapping
+from typing import Optional, Sequence, Mapping, Type, Any
 
 KeyPhrase = tuple[str, ...]
 KeyPhrases = Sequence[KeyPhrase]
 
 
+def check_type_and_emptiness(check_object: Any, check_type: Type[Any], can_be_empty: bool = True) -> bool:
+    """
+    Checks if object's type matches required type.
+    Checks if the object is empty if necessary.
+    """
+    if not isinstance(check_object, check_type):
+        return False
+    if not check_object and not can_be_empty:
+        return False
+    return True
+
+
+def check_list(check_object: Any, token_type: Type[Any], can_be_empty: bool = True) -> bool:
+    """
+    Checks if an object is a list and checks types of its elements.
+    Checks if the object is empty if necessary.
+    """
+    if not check_type_and_emptiness(check_object, list, can_be_empty):
+        return False
+    for i in check_object:
+        if not isinstance(i, token_type):
+            return False
+    return True
+
+
+def check_dict(check_object: Any, key_type: Type[Any], value_type: Type[Any],
+               can_be_empty: bool = True) -> bool:
+    """
+    Checks if an object is a dictionary and checks types of its keys and values.
+    Checks if the object is empty if necessary.
+    """
+    if not check_type_and_emptiness(check_object, dict, can_be_empty):
+        return False
+    for key, value in check_object.items():
+        if not isinstance(key, key_type) or not isinstance(value, value_type):
+            return False
+    return True
+
+
+def check_tuple(check_object: Any, token_type: Type[Any], can_be_empty: bool = True) -> bool:
+    """
+    Checks if an object is a tuple and checks types of its elements.
+    Checks if the object is empty if necessary.
+    """
+    if not check_type_and_emptiness(check_object, tuple, can_be_empty):
+        return False
+    for i in check_object:
+        if not isinstance(i, token_type):
+            return False
+    return True
+
+
+def check_int(check_object: Any) -> bool:
+    """
+    Checks if an object is a positive integer.
+    """
+    if not isinstance(check_object, int) or isinstance(check_object, bool):
+        return False
+    if not check_object > 0:
```

Comment:
what does this check do? |
It is too difficult to understand. 
Can you simplify?

Reply:
removed it (it isn't needed yet), will simply import from the first lab if needed